### PR TITLE
Implementation of gas limits for relaying messages to Linea

### LINKED ISSUE #3: dev: Determine typical gas limits for relaying messages to Linea
### Description

We need to calculate and set appropriate gas limits for relaying messages from Ethereum to Linea in our World ID State Bridge contract.

### Tasks

1. Find a method to calculate gas needed for relaying messages from Ethereum to Linea.
2. Estimate gas limits required for each type of message sent from the state bridge on Ethereum to the World ID contract on Linea.
3. Modify the [LineaStateBridge](https://github.com/kfastov/worldcoin-bridge-linea/blob/main/src/LineaStateBridge.sol) contract to include default gas limits for each message type.

### Acceptance Criteria

- [ ] Gas calculation method is documented
- [ ] Gas estimates for each message type are provided and justified.
- [ ] [LineaStateBridge](https://github.com/kfastov/worldcoin-bridge-linea/blob/main/src/LineaStateBridge.sol) contract is updated with appropriate gas limit constants or variables.
- [ ] Code changes are thoroughly tested and reviewed.

### Resources

- [Linea Documentation](https://docs.linea.build/)

### PR DESCRIPTION
**Pull request by Ugo-X**
This PR introduces three new functions to manage gas limits for critical cross-chain operations on the Linea blockchain. These functions allow the contract owner to dynamically adjust gas limits for specific methods, enhancing the flexibility and efficiency of our cross-chain communication system.

Changes:
1. Added `setGasLimitPropagateRoot` function to set the gas limit for the `propagateRoot` method.
2. Added `setGasLimitSetRootHistoryExpiry` function to set the gas limit for the `setRootHistoryExpiry` method.
3. Added `setGasLimitTransferOwnership` function to set the gas limit for the `transferOwnership` method.

closes #3

### CODE REVIEW

**Comment by kfastov** [comment]
@iamUgo_X Hi! Sorry for making you wait for so long. This issue was mostly theoretical and required some research to determine some initial values for gas limits. That's cool though that you also added the code for setting those limits. I now realize that description was not so clear, and that assigning this task before contracts are ready was a bit dishonest, and I've started to review it too late, so I won't ask you to change anything.
But can you please run `forge fmt` on the code to satisfy the linter? 😊

**Comment by Ugo-X** [comment]
> @iamUgo_X Hi! Sorry for making you wait for so long. This issue was mostly theoretical and required some research to determine some initial values for gas limits. That's cool though that you also added the code for setting those limits. I now realize that description was not so clear, and that assigning this task before contracts are ready was a bit dishonest, and I've started to review it too late, so I won't ask you to change anything. But can you please run `forge fmt` on the code to satisfy the linter? 😊

it's no issue, thanks chief, I have run forge fmt and updated.

**Comment by kfastov** [comment]
I'll merge this PR after we merge the contracts because they have some common code

**Comment by Ugo-X** [comment]
Alright chief, will get to that.

**Comment by Ugo-X** [comment]
> @Ugo-X Since contracts are merged now, can you please try to determine the reasonable default gas limits for those methods? You may deploy them on local devnet or testnet or use some online tools like Remix. Don't hesitate to contact me if you need any help

Am I to calculate the exact gas fees?

**Comment by Ugo-X** [comment]
Alright

**Comment by kfastov** [comment]
@Ugo-X Okay I'll try to get some testing ETH on Sepolia and share with you 🙏

**Comment by Ugo-X** [comment]
Alright,thanks 🙏

File: src/LineaStateBridge.sol
Comment (kfastov) [inline src/LineaStateBridge.sol:40]: Can you please share how did you get this value? You can provide some screenshots, logs, or just describe it in a few words

Comment (Ugo-X) [reply src/LineaStateBridge.sol:40]: > Can you please share how did you get this value? You can provide some screenshots, logs, or just describe it in a few words

I deployed the contract and used the contract deployed gas limit as the default gas limit.

Screenshot 2024-07-05 at 22 06 00
Screenshot 2024-07-05 at 22 06 10

LINK:  https://sepolia.lineascan.build/tx/0xd6ddc75572298d7d6c69ab9bc21227ef45067856016bcde294f6f4f166ab19e5

File: bun.lockb
Comment (kfastov) [inline bun.lockb:1]: Hmm that lock file changes constantly, and even its attributes do. I should read the bun's best practices 🤔

File: src/LineaStateBridge.sol
Comment (kfastov) [reply src/LineaStateBridge.sol:40]: @Ugo-X Thanks for deploying the contracts on testnet and sharing all the details!
However it seems that you estimated gas limit for the deploy transaction, not the method calls.
Also, I think `1_000_000` is a good first approximation for `988_315`. We'll figure out the actual values when we will test the contracts on testnet

Comment (kfastov) [inline src/LineaStateBridge.sol:40]: I propose to just leave it as is and add a comment reminding of this work
```suggestion
    // estimated by @Ugo-X as ~988_315
    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
```

**Review by kfastov — CHANGES_REQUESTED** [review]
Approve with comments above

**Comment by Ugo-X** [comment]
reverted back to the original Approximate gas limit. How should i proceed?

**Pull request merged**
